modules/platforms/dotnet/Apache.Ignite.Core/Services/IServiceContext.cs

```diff
@@ -65,5 +66,13 @@ public interface IServiceContext
         /// Affinity key, possibly null.
         /// </value>
         object AffinityKey { get; }
+
+        /// <summary>
+        /// Gets context of the current service call. 
+        /// </summary>
+        /// <returns>Context of the current service call.</returns>
+        /// <seealso cref="IServiceCallContext"/>
+        [IgniteExperimental]
+        IServiceCallContext CurrentCallContext();
```

Comment:
Let's make it a property, more idiomatic:

```suggestion
        IServiceCallContext CurrentCallContext { get; }
```

Reply:
Thanks, the "getter" method has been changed.

But for the "setter" method I see 3 options:
1. Leave the static method SetCurrentCallContext(..) in the ServiceContext (current solution). 
2. Make it instance method (normal "setter").
3. Extract the threadlocal to another container-class with static methods to reading/storing the context (similar to current java implementation).

Personally, I prefer the second one, but accessing the ServiceContext instance while executing a service method requires a bit of reworking of the current service logic. 

Personally, I prefer the second one, but it requires a little rework of current service logic to access ServiceContext instance when executing the service method:
a. Store ServiceContext instead of a service in handleregistry.
b. Store reference to the Service in ServiceContext implementation.

Reply:
Setter is internal, I think it is fine to keep it as is. No need to complicate.

Reply:
ok, thanks for the reply.